addons/sale/models/sale_order_line.py

```diff
@@ -384,7 +384,7 @@ def _compute_product_uom(self):
             if not line.product_uom or (line.product_id.uom_id.id != line.product_uom.id):
                 line.product_uom = line.product_id.uom_id
 
-    @api.depends('product_id', 'company_id')
+    @api.depends('product_id', 'company_id', 'order_id.fiscal_position_id')
```

Comment:
We do not want to force the recomputation of the taxes everytime the fiscal position is modified (at least not in backend).  That's why we have the whole `show_update_fpos` logic.
It should be handled in website_sale when needed (cf fixup commit).